### ISSUE TITLE
[Regression] Unable to run fastlane v2.3.1 - can't satisfy 'json (< 3.0.0, >= 2.0.1)'

### ISSUE THREAD

**Issue by jinjorge** [fastlane/fastlane#7696]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I have read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)

##### Complete output when running fastlane, including the stack trace and command used
> You can use: `--capture_output` as the last commandline argument to get that collected for you

[INSERT OUTPUT HERE]

### Environment

Please run `fastlane env` and copy the output below. This will help us help you :+1:
If you used `--capture_output` option please remove this block - as it is already included there.

With v2.3.1 all fastlane cmds fails with:

```
✗ fastlane --version
/Users/jeremiah/.rvm/rubies/ruby-2.3.1/lib/ruby/2.3.0/rubygems/specification.rb:1435:in `block in activate_dependencies': can't satisfy 'json (< 3.0.0, >= 2.0.1)', already activated 'json-1.8.3' (Gem::LoadError)
	from /Users/jeremiah/.rvm/rubies/ruby-2.3.1/lib/ruby/2.3.0/rubygems/specification.rb:1427:in `each'
	from /Users/jeremiah/.rvm/rubies/ruby-2.3.1/lib/ruby/2.3.0/rubygems/specification.rb:1427:in `activate_dependencies'
	from /Users/jeremiah/.rvm/rubies/ruby-2.3.1/lib/ruby/2.3.0/rubygems/specification.rb:1409:in `activate'
	from /Users/jeremiah/.rvm/rubies/ruby-2.3.1/lib/ruby/2.3.0/rubygems/core_ext/kernel_gem.rb:68:in `block in gem'
	from /Users/jeremiah/.rvm/rubies/ruby-2.3.1/lib/ruby/2.3.0/rubygems/core_ext/kernel_gem.rb:67:in `synchronize'
	from /Users/jeremiah/.rvm/rubies/ruby-2.3.1/lib/ruby/2.3.0/rubygems/core_ext/kernel_gem.rb:67:in `gem'
	from /Users/jeremiah/.rvm/gems/ruby-2.3.1/bin/fastlane:22:in `<main>'
	from /Users/jeremiah/.rvm/gems/ruby-2.3.1/bin/ruby_executable_hooks:15:in `eval'
	from /Users/jeremiah/.rvm/gems/ruby-2.3.1/bin/ruby_executable_hooks:15:in `<main>'
```

Here's the `fastlane env` with v2.3.0. 

<details><summary>🚫 fastlane environment 🚫</summary>

### Stack

| Key                 | Value                                       |
| ------------------- | ------------------------------------------- |
| OS                  | 10.12.3                                     |
| Ruby                | 2.3.1                                       |
| Bundler?            | false                                       |
| Git                 | git version 2.10.1 (Apple Git-78)           |
| Installation Source | ~/.rvm/gems/ruby-2.3.1/bin/fastlane         |
| Host                | Mac OS X 10.12.3 (16D17a)                   |
| Ruby Lib Dir        | ~/.rvm/rubies/ruby-2.3.1/lib                |
| OpenSSL Version     | OpenSSL 1.0.2j  26 Sep 2016                 |
| Is contained        | false                                       |
| Xcode Path          | /Applications/Xcode.app/Contents/Developer/ |
| Xcode Version       | 8.2                                         |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   |             |   |
| LANGUAGE |             |   |

### fastlane files:

**No Fastfile found**

**No Appfile found**

### fastlane gems

| Gem      | Version | Update-Status       |
| -------- | ------- | ------------------- |
| fastlane | 2.3.0   | 🚫 Update availaible |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| did_you_mean              | 1.0.0        |
| executable-hooks          | 1.3.2        |
| bundler-unload            | 1.0.2        |
| rubygems-bundler          | 1.4.4        |
| bundler                   | 1.13.6       |
| io-console                | 0.4.5        |
| slack-notifier            | 1.5.1        |
| i18n                      | 0.7.0        |
| json                      | 1.8.3        |
| thread_safe               | 0.3.5        |
| tzinfo                    | 1.2.2        |
| minitest                  | 5.8.3        |
| activesupport             | 4.2.7.1      |
| CFPropertyList            | 2.3.3        |
| claide                    | 1.0.1        |
| colored                   | 1.2          |
| nanaimo                   | 0.2.2        |
| xcodeproj                 | 1.4.1        |
| rouge                     | 1.11.1       |
| xcpretty                  | 0.2.4        |
| terminal-notifier         | 1.7.1        |
| terminal-table            | 1.4.5        |
| plist                     | 3.2.0        |
| addressable               | 2.3.8        |
| multipart-post            | 2.0.0        |
| word_wrap                 | 1.0.0        |
| babosa                    | 1.0.2        |
| highline                  | 1.7.8        |
| commander                 | 4.4.0        |
| excon                     | 0.54.0       |
| faraday                   | 0.10.0       |
| unf_ext                   | 0.0.7.2      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20161021 |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| fastimage                 | 1.6.8        |
| gh_inspector              | 1.0.2        |
| uber                      | 0.0.15       |
| representable             | 2.3.0        |
| retriable                 | 2.1.0        |
| mime-types-data           | 3.2016.0521  |
| mime-types                | 3.1          |
| hurley                    | 0.2          |
| little-plugger            | 1.1.4        |
| multi_json                | 1.12.1       |
| logging                   | 2.1.0        |
| jwt                       | 1.5.6        |
| memoist                   | 0.15.0       |
| os                        | 0.9.6        |
| signet                    | 0.7.3        |
| googleauth                | 0.5.1        |
| httpclient                | 2.8.2.4      |
| google-api-client         | 0.9.20       |
| mini_magick               | 4.5.1        |
| multi_xml                 | 0.5.5        |
| security                  | 0.1.3        |
| xcpretty-travis-formatter | 0.0.4        |
| dotenv                    | 2.1.1        |
| faraday_middleware        | 0.10.1       |
| rubyzip                   | 1.2.0        |
</details>

*generated on:* **2016-12-28**
</details>

The `gem list --local` output shows two version of the `json` gem installed

```
✗ gem list json --local

*** LOCAL GEMS ***

json (2.0.2, 1.8.3)
multi_json (1.12.1)
```
Looks like the [change](https://github.com/fastlane/fastlane/pull/7674) made to satisfy Ruby 2.4 may be causing a problem for non Ruby 2.4 systems.

Downgrading to fastlane v2.3.0 all works well

**Comment by itchingpixels** [comment]
Same here! will try to upgrade to ruby 2.4 now.

**Comment by carlosefonseca** [comment]
I think you might be able to uninstall version 1.8.3 and it might run. I believe there's still some stuff like xcversion that seems to be failing with newer json.

**Comment by jinjorge** [comment]
@carlosefonseca can't uninstall v1.8.3 as it's part of the `system` gems

**Comment by carlosefonseca** [comment]
@jinjorge sometimes it's also installed as a normal gem and when that one is removed, things start working. But I try different things because shit breaks and it's not that easy to know why or what got it fixed…

**Comment by jinjorge** [comment]
@carlosefonseca on the build machines that I run into this issues, v1.8.3 was the system gem and v2.0.2 was installed under the user gems

**Comment by math-nao** [comment]
Same problem here. I solved it by updating json package to `2.0.2`.

**Comment by 365SplendidSuns** [comment]
I'm having the same problem. Anyone have a working suggestion to fix this? Thanks guys.

**Comment by haydus** [comment]
I found a temp workaround that works for me:
1. uninstall fastlane
2. Re-install fastlane - it will install fastlane 2.3.0
3. Things will work again now although with previous version of fastlane, but you can't also update fastlane to 2.3.1 but don't run gem cleanup.
For some reason only after running 'gem cleanup' fastlane starts complaining about json 1.8.3

Hopes this helps.

**Comment by hjanuschka** [comment]
hey folks i think those issues have been resolved - any one can confirm this?

**Comment by jinjorge** [comment]
@hjanuschka yes, this is resolved. Thanks!
